src/flint/types/fmpz_mod.pyx

```diff
@@ -0,0 +1,340 @@
+from flint.flintlib.flint cimport slong
+from flint.flintlib.fmpz cimport (
+    fmpz_t,
+    fmpz_one,
+    fmpz_set,
+    COEFF_IS_MPZ,
+    fmpz_get_str,
+    fmpz_init,
+    fmpz_clear,
+    fmpz_equal
+)
+from flint.flintlib.fmpz_mod cimport *
+
+from flint.utils.typecheck cimport typecheck
+from flint.utils.conversion cimport str_from_chars
+from flint.flint_base.flint_base cimport flint_scalar
+from flint.types.fmpz cimport (
+    fmpz,
+    any_as_fmpz,
+)
+
+cimport libc.stdlib
+
+# TODO: import this from types.fmpz somehow, it's not good
+# to have the function repeated here.
+cdef fmpz_get_intlong(fmpz_t x):
+    """
+    Convert fmpz_t to a Python int or long.
+    """
+    cdef char * s
+    if COEFF_IS_MPZ(x[0]):
+        s = fmpz_get_str(NULL, 16, x)
+        v = int(str_from_chars(s), 16)
+        libc.stdlib.free(s)
+        return v
+    else:
+        return <slong>x[0]
+
+cdef class fmpz_mod_ctx:
+    """
+    """
+    def __cinit__(self, mod):
+        self.__init__(mod)
```

Comment:
`__cinit__` should not call `__init__`. The `__cinit__` function is called as part of `__new__` to create the object. The `__init__` method will be called later.

Here is where `fmpz_mod_ctx_init` should be called.